テストを用いたAPIの挙動の確認

takahoo opened the pull request:
## What
<!--- Write the change being made with this pull request --->
Step6-2 Hello Handlerのテストを書く  
Step6-4実際のデータベースを用いたテストを書く  

##Why
- 期待する出力200と{message: Hello world!}が得られるかを確認
- 正しいレスポンスを返すことの確認

##実装結果
- errorが何度も出ていて、main_tesy.pyを修正していたが、main.pyの影響であったため、main.pyを修正し、FastAPIが正しいdictionaryを返すように

```
(venv) takaho.ysz@MacBook-Air-3 python % pytest
============================= test session starts ==============================
platform darwin -- Python 3.13.0, pytest-8.3.5, pluggy-1.5.0
rootdir: /Users/takaho.ysz/Desktop/mercari_build_2025/mercari-build-training/python
plugins: anyio-4.8.0
collected 3 items                                                              

main_test.py ...                                                         [100%]

============================== 3 passed in 0.21s ===============================
```

## CHECKS :warning:

Please make sure you are aware of the following.

- [ ] **The changes in this PR doesn't have private information

takahoo committed:
コメントアウト

dominic-soh commented on python/main.py:
```diff
@@ -115,44 +145,39 @@ async def add_item(
 async def get_image(image_name:str):
     # Create image path
-    image = images / image_name
+    image = Image_dir / image_name
```
I think you can reference `images` from line 21 for the directory.

dominic-soh commented on python/main.py:
```diff
@@ -115,44 +145,39 @@ async def add_item(
     if not image.exists():
         logger.debug(f"Image not found: {image}")
-        image = images / "default.jpg"
+        image = Image_dir / "default.jpg"
```
Here as well

takahoo committed:
Update main.py

takahoo committed:
Add items.json

takahoo committed:
git push origin sixth-pull-request
Merge remote-tracking branch 'origin/sixth-pull-request' into sixth-pull-request

kazuki-hanai commented on python/items.json:
このファイルは要らない？🤔

kazuki-hanai commented on python/main.py:
```diff
@@ -62,38 +82,48 @@ async def lifespan(app: FastAPI):
-    image_name: str
-
-def insert_item(item: Item):
+def insert_item(item: Item, db:sqlite3.Connection):
```
```suggestion
def insert_item(item: Item, db: sqlite3.Connection):
```

みやすさのためにスペース入れましょう

kazuki-hanai reviewed (commented):
レビューしました！

takahoo committed:
Delete python/items.json

takahoo committed:
Merge branch 'main' into sixth-pull-request

kazuki-hanai reviewed (approved):
![LGTM](http://i.giphy.com/3o7TKrp0EjTP6FMHnO.gif)
